compiler/kernel/vop.go

```diff
@@ -176,6 +176,12 @@ func (b *Builder) compileVamLeaf(o dag.Op, parent vector.Puller) (vector.Puller,
 			return nil, err
 		}
 		return vamop.NewYield(b.zctx(), parent, []vamexpr.Evaluator{e}), nil
+	case *dag.DefaultScan:
```

Comment:
Won't this change make it so CSUP and Parquet are materialized then dematerialized before hitting the vector runtime?

Reply:
Yes, but it's a step forward from this.

```console
$ SUPER_VAM=1 super f.csup
vector runtime does not support internal readers
```

Fixing things so it we skip that extra work requires some refactoring of the way files on the command line are handled, and I look forward to doing that.